### ISSUE TITLE
RSpec/UnspecifiedException triggers for `not_to` within a `to` block

### ISSUE THREAD

**Issue by G-Rath** [rubocop/rubocop-rspec#1954]
Given the following:

```
expect do
  expect { described_class.new(report).populate! }.not_to raise_error
end.to change(ReportTicket, :count).by(3)
```

Before #1940 this was fine, but now it's being flagged as an error - if I replace the outer `to` with `not_to`, it passes again.

**Comment by pirj** [comment]
Walking up the tree seems incorrect [here](https://github.com/rubocop/rubocop-rspec/blob/74d78371f4495099cf23bcf93b8af7ef711aec26/lib/rubocop/cop/rspec/unspecified_exception.rb#L65).
I recall making a very similar note in another issue [here](https://github.com/rubocop/rubocop-rspec/issues/1949#issuecomment-2285070419), and the fix is the same.

@rubocop/rubocop-rspec why do we have this cop? Doesn’t it fall into a non-goal category considering [this RSpec built-in warning](https://github.com/rspec/rspec-expectations/blob/2bf42aaa1621202f0fb71b371dbf6c3fec1f4291/lib/rspec/matchers/built_in/raise_error.rb#L189)?

Should we fix the cop, or deprecate it?

**Comment by bquorning** [comment]
It looks like that feature [was added](https://github.com/rspec/rspec-expectations/commit/e70959a13a5a382dcfec2a810f75b6c3450d955a) in RSpec 3.3, over 9 years ago. I’d say we should deprecate this cop.

**Comment by ydah** [comment]
It might still have been worthwhile to have this cop if it had supported autocorrection, but since it does not support autocorrection, this cop should be deprecated.

**Comment by G-Rath** [comment]
hmm so on second thought, I'm less in favor of deprecating this cop now because it seems RSpec doesn't fail when printing this warning - it might be possible to configure RSpec to error, but overall it does seem slightly useful to be able to lint this in parallel with running your whole test suite.

It also sounds like there's someone working [to address this](https://github.com/rubocop/rubocop-rspec/issues/1949#issuecomment-2341284685) - I assume that while the team have expressed a wish to deprecate the rule, they still wouldn't outright reject a pull request fixing the underlying issue so I might just for now add an inline disable in my project for now and see how this unfolds in the community
